### DISCUSSION TITLE
Comparison with Denote and its "signature" feature

### DISCUSSION THREAD

**Discussion by gour** [ahyatt/ekg#60] [Q&A]
Hello,

I've tried org-roam, then found out about Denote...in the meantime played a bit with Logseq, but not having need for mobile note-taking, decided to settle on Emacs-based solution.

Denote also has got "signature" feature which enables one to mimic Luhmann-like index-card naming.

I've watched Andrew' YT video being aware he does not have experience with Denote, I wonder if some other `ekg` users can share their comparison with the two packages?

One thing is certain - I'm going to try it out and I'm sure it can replace my need for [nb](https://github.com/xwmx/nb), but I wonder if it can serve my needs for Zettelkasten/PKM as well? ("Tags instead of backlinks" as well as sqlite3 storage are great features.)

**Comment by jayrajput** [comment]
I am playing with both `denote` and `ekg`. Mostly it is `ekg`, but as and when I need bigger files, I end up create it with `denote` and linking it to my `ekg` notes using `ekg-capture-url`. 

I am also reading "Building a Second Brain" by Tiago Forte and trying to implement ideas in my note taking. 

I have played in past with `org-roam` more than a year. I have used back links there, but I have gone away with the need of using backlinks. Using common tags across notes is a way of creating backlinks. Although ekg do support ways to link to other notes, I have not seen a use of that in my real life.

The `ekg` concept of capturing URL/resource and taking notes against them is life changing. Unless and until I have something to say about a URL, I do not try to capture it in my notes, if I know I can find it from somewhere else. Learning the BASB book, I am more cautious about what enters my notes. 

The `ekg` is also implicitly forcing me to create smaller/atomic notes which is much appreciate in all the note taking to create smaller/atomic notes.

I have also start reviewing my notes more and more. Tags help with that a lot. With `org-roam`, the review workflow was not this natural.

So overall `ekg` is awesome and planning to continue using it and see how it goes.

Further the code of `ekg` is quite easy to follow. It may grow. But as of now, you can easily follow the code.

Let me know if you have specific questions. I will be happy to answer/discuss the same.

**Reply by gour** [reply]
Thank you very much for your input!

> Let me know if you have specific questions. I will be happy to answer/discuss the same.

I'm curious what do you think about Denote's "signature" feature and/or Luhmann-like naming scheme, iow. 1,1a,2,2a,2b etc. which provides some historical trace of thought or context how one has arrived to certain thought/note?

**Reply by jayrajput** [reply]
I have personally not played with the signature feature. I did played a lot with the backlinks, but it did not worked for me to be candid. It was fun, I can see all the backlinks in the `org-roam-ui` and what not. But somehow it did not worked for me. Your mileage may wary.

So note taking has two reasons
1. Able to find my notes quicker when needed.
2. Automatically develop insights by these backlinks.

Number 2 is what I was initially targeting when using `org-roam`, but kind of failed. Now my primary goal is Number 1 and `ekg` works great for that. My thinking has also molded where I am thinking of information in terms of tagging. 

To summarize, I do not have insights into Denote signature feature, but I a not using backlink in my workflow. Mostly tagging is what I am using. I do link `ekg` notes to flat files like pptx, pdf, org, txt, zip, etc.

**Reply by gour** [reply]
Hello,

please excuse me for a late reply...I was changing distros and being busy with some other computer-non-related stuff...

> I have personally not played with the signature feature. I did played a lot with the backlinks, but it did not worked for me to be candid. It was fun, I can see all the backlinks in the `org-roam-ui` and what not. But somehow it did not worked for me. Your mileage may wary.

I can suggest to you to take a look at [this](https://www.thesing-online.de/blog/denote-as-a-zettelkasten) video series which does a good job of explaining possible value of "signatures"...

 
> Number 2 is what I was initially targeting when using `org-roam`, but kind of failed. Now my primary goal is Number 1 and `ekg` works great for that. My thinking has also molded where I am thinking of information in terms of tagging.

...and is nice explanation for your point Numer 1. ;)

Let me know what do you think?

**Reply by jayrajput** [reply]
I will need to reserve some time to view the videos. But I am not a video guy :-). I can read text, but never videos for some reasons.

**Comment by jayrajput** [comment]
I am not sure if you have tried embeddings feature of ekg. I am playing with it and as I understand it will use Open AI (Chat GPT) to produce extra text/details/insights for your notes. This kind of take care of the number 2 reasons for which one take notes. I am still setting up things for embeddings though.

**Reply by gour** [reply]
> I am not sure if you have tried embeddings feature of ekg. 

Not yet.

> I am playing with it and as I understand it will use Open AI (Chat GPT) to produce extra text/details/insights for your notes. This kind of take care of the number 2 reasons for which one take notes. I am still setting up things for embeddings though.

As far as I'm concerned, I do rather not depend on AI to get insight into my notes. :smile_cat:

**Reply by jayrajput** [reply]
Privacy is definitely a concern.

**Comment by jayrajput** [comment]
Btw, I started writing ekg-denote.el to integrate EKG and Denote. By integrating, I mean that you can import/exprot between the two systems. I still need to get the modules reviewed and tested though. The code is here: https://github.com/jayrajput/ekg/blob/ekg-denote/ekg-denote.el. This is just a proof of flexibility of both the systems (Ekg and Denote). Btw, Andrew wrote an importer/export for logseq. I just copied it and adjusted it for denote. The power of Ekg comes from the Database. The package itself is quite simple due to triples which does all the heavy lifting.

**Comment by gour** [comment]
> Btw, I started writing ekg-denote.el to integrate EKG and Denote. By integrating, I mean that you can import/exprot between the two systems.

This is very interesting...I wonder whether `ekg-denote.el` can handle signatures in Denote's files?

> The power of Ekg comes from the Database. The package itself is quite simple due to triples which does all the heavy lifting.

Yes, I'm very enthusiastic about it...files are nice, but (sqlite3) database can make many things much simpler and/or quicker...Fossil is my favorite DVCS. :wink:
